pkg/compose/plugins.go

```diff
@@ -61,15 +61,33 @@ func (s *composeService) runPlugin(ctx context.Context, project *types.Project,
 
 	cmd := s.setupPluginCommand(ctx, project, provider, plugin.Path, command)
 
+	variables, err := s.executePlugin(ctx, cmd, command, service)
+	if err != nil {
+		return err
+	}
+
+	prefix := strings.ToUpper(service.Name) + "_"
+	for name, s := range project.Services {
+		if _, ok := s.DependsOn[service.Name]; ok {
+			for key, val := range variables {
```

Comment:
micro-nit 😅 (definitely not blocking); looks like the `prefix` is only used when `depends_on` matches, so could be put inside that branch; 

```suggestion
	for name, s := range project.Services {
		if _, ok := s.DependsOn[service.Name]; ok {
			prefix := strings.ToUpper(service.Name) + "_"
			for key, val := range variables {
```